examples/Python/ChatApp/app.py

```diff
@@ -0,0 +1,192 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""
+Azure OpenAI Chat Application using Azure App Configuration.
+This script demonstrates how to create a chat application that uses Azure App Configuration
+to manage settings and Azure OpenAI to power chat interactions.
+"""
+
+import os
+from typing import List, Dict, Any
+from azure.core.credentials import TokenCredential
+from azure.identity import DefaultAzureCredential, get_bearer_token_provider
+from azure.appconfiguration.provider import load, SettingSelector, WatchKey
+from openai import AzureOpenAI
+from models import AzureOpenAIConfiguration, Message, ModelConfiguration
+
+
+def main():
+    # Create a credential using DefaultAzureCredential
+    credential = DefaultAzureCredential()
+    refresher = None
+
+    # Get the App Configuration endpoint from environment variables
+    app_config_endpoint = os.environ.get("AZURE_APPCONFIGURATION_ENDPOINT")
+    if not app_config_endpoint:
+        raise ValueError(
+            "The environment variable 'AZURE_APPCONFIGURATION_ENDPOINT' is not set or is empty."
+        )
+
+    try:
+        azure_openai_config = None
+        azure_client = None
+
+        def on_refresh_success():
+            azure_openai_config, azure_client = _create_ai_client(provider)
```

Comment:
I'd expect something like this
```python
        def on_refresh_success():
            azure_openai_config = _get_azure_openai_config(appconfig)
            chat_completion_config = _get_chat_completion_configuration(appconfig)
            azure_openai_client = _create_azure_openai_client(azure_openai_config)
```

Reply:
The class made the code hard to read. I have to jump back and forth to follow the flow.